### ISSUE TITLE
sbt plugin not compatible with sbt/zinc 1.4.x as it uses internal classes

### ISSUE THREAD

**Issue by jtjeferreira** [playframework/playframework#10497]
### Play Version

2.7.4 (but I think it should also happen in other 2.7.x or 2.8.x)

### API

Scala

### Operating System

5.8.16-2-MANJARO

### JDK

AdoptOpenJDK 11.0.8

### Library Dependencies

sbt 1.4.1

### Expected Behavior

When a runtime error happens in a play application running in sbt 1.4.x I would expect to see the error like in when running with sbt 1.3.x.

### Actual Behavior

Instead I see this exception:

```
java.lang.ClassCastException: class sbt.internal.inc.MappedVirtualFile cannot be cast to class java.io.File (sbt.internal.inc.MappedVirtualFile is in unnamed module of loader sbt.internal.MetaBuildLoader @38afe297; java.io.File is in module java.base of loader 'bootstrap')
	at play.sbt.run.PlayReload$.$anonfun$sourceMap$1(PlayReload.scala:69)
	at scala.collection.MapLike$MappedValues.get(MapLike.scala:260)
	at play.runsupport.Reloader.$anonfun$findSource$1(Reloader.scala:547)
	at scala.Option.flatMap(Option.scala:271)
	at play.runsupport.Reloader.findSource(Reloader.scala:546)
	at play.core.server.DevServerStart$$anon$1$$anon$2.sourceOf(DevServerStart.scala:172)
	at play.core.SourceMapper.$anonfun$sourceFor$1(ApplicationProvider.scala:21)
	at play.core.SourceMapper.$anonfun$sourceFor$1$adapted(ApplicationProvider.scala:21)
	at scala.collection.IndexedSeqOptimized.$anonfun$find$1(IndexedSeqOptimized.scala:53)
	at scala.collection.IndexedSeqOptimized.$anonfun$find$1$adapted(IndexedSeqOptimized.scala:53)
```

### Reproducible Test Case

Still working on it

**Comment by danellis** [comment]
Dupe of #10486

**Comment by jtjeferreira** [comment]
> Dupe of #10486

@danellis true

**Comment by mkurz** [comment]
Fix for master: #10648
Fix for 2.8.x: #10649
Fix for 2.7.x: #10650

**Comment by PromanSEW** [comment]
I updated Play to 2.8.11, SBT to 1.5.5 and still get this error (JDK 1.8.0.301)
When this error will be fixed?
All versions after 2.8.7 with SBT 1.4.0+ have this error, I can't upgrade my project so long
